src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double width;
+    private double height;
+
+    public Rectangle(Color color, double width, double height) {
+        super(color);
+        this.width = width;
+        this.height = height;
+    }
+
+    @Override
+    public double getArea() {
+        return width * height;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: rectangle, area: " + getArea() + " sq. units, width: "
+                + width + " units, height: " + height + " units, color: "
+                + new ColorSupplier().getRandomColor());
+    }

```

Comment:
The class `Rectangle` should not extend `Figure` if `Figure` is an abstract class meant to set behavior. Consider using an interface to define shared behavior, and use composition over inheritance if there's shared state or behavior that doesn't fit an is-a relationship.